src/VisualStudio/Core/Def/RoslynPackage.cs

```diff
@@ -39,9 +40,14 @@ internal class RoslynPackage : AbstractPackage
         private RuleSetEventHandler _ruleSetEventHandler;
         private IDisposable _solutionEventMonitor;
 
-        protected override void Initialize()
+        protected override async Task InitializeAsync(CancellationToken cancellationToken, IProgress<ServiceProgressData> progress)
         {
-            base.Initialize();
+            await base.InitializeAsync(cancellationToken, progress).ConfigureAwait(true);
+
+            await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+
+            var componentModel = (IComponentModel)await GetServiceAsync(typeof(SComponentModel)).ConfigureAwait(true);
+            cancellationToken.ThrowIfCancellationRequested();
 
```

Comment:
Add `Assumes.Present(componentModel);` to generate a more actionable exception than NRE later on.

Reply:
➡️ Fixed in e185f963baab126dd45e6ff6b7a768d9263660af